app/services/facet.service.js

```diff
@@ -86,17 +86,20 @@
     var filter = function(values) {
       var self = this; // needs to be defined outside the filter function
       return values.filter(function(element, index, array) {
-        var result;
-        for (var key in element) {
-          var value = element[key];
-          result = filter_value.apply(self, [key, value]);
-          if (result === false) {
-            return false;
-          }
-        }
+        return self.passes_filters(element);
+      });
+    };
 
-        return true;
+    var passes_filters = function(object) {
+      var self = this;
+      var result = true;
+      angular.forEach(object, function(value, key) {
+        if (filter_value.apply(self, [key, value]) === false) {
+          result = false;
```

Comment:
Updated to iterate over `Object.keys`, which [according to MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys) does do `hasOwnProperty` filtering. Added it as a separate commit since it's a change to the logic (not moving existing logic to a new place like above).